src/components/content.js

```diff
@@ -698,15 +699,25 @@ class Content extends React.Component {
     if (this.props.readOnly) return
     if (!this.isInEditor(event.target)) return
 
-    event.preventDefault()
+    const handleData = (data) => {
+      // Attach the `isShift` flag, so that people can use it to trigger "Paste
+      // and Match Style" logic.
+      data.isShift = !!this.tmp.isShifting
+
+      debug('onPaste', { event, data })
+      this.props.onPaste(event, data)
+    }
+
     const data = getTransferData(event.clipboardData)
 
-    // Attach the `isShift` flag, so that people can use it to trigger "Paste
-    // and Match Style" logic.
-    data.isShift = !!this.tmp.isShifting
+    const doesNotSupportHtmlFromClipboard = IS_IE // May be extended with IS_EDGE and IS_SAFARI...?
 
-    debug('onPaste', { event, data })
-    this.props.onPaste(event, data)
+    if (doesNotSupportHtmlFromClipboard) {
```

Comment:
Can you change this to just `if (IS_IE) {`?

And then can you add a comment about explaining exactly why we need this code, so that for future editing we don't break it. (Search the codebase for `COMPAT:` comments to see examples of how much detail to use, and include today's date.)